src/main/java/edu/ucla/library/iiif/auth/Config.java

```diff
@@ -131,11 +141,45 @@ public final class Config {
      */
     public static final String SINAI_COOKIE_VALID_PREFIX = "SINAI_COOKIE_VALID_PREFIX";
 
+    /**
+     * A logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(Config.class, MessageCodes.BUNDLE);
+
     /**
      * Constant classes should have private constructors.
      */
     private Config() {
         // This is intentionally left empty
     }
 
+    /**
+     * A configuration processor that adds the application version, if available.
+     *
+     * @param aConfig An application configuration
+     * @return The processed application configuration
+     */
+    public static JsonObject setAppVersion(final JsonObject aConfig) {
+        final String manifestPath = "/META-INF/MANIFEST.MF";
+
+        try (InputStream manifest = Config.class.getResourceAsStream(manifestPath)) {
+            final Properties properties = new Properties();
+            final Optional<String> version;
+
+            properties.load(manifest);
+            version = Optional.ofNullable(properties.getProperty("Maven-Version"));
+
+            if (version.isPresent()) {
+                return aConfig.copy().put(HAUTH_VERSION, version.get());
+            } else {
+                return aConfig;
+            }
+        } catch (final IOException details) {
+            // Either the app wasn't deployed as a JAR, or Vert.x Maven Plugin isn't creating the manifest file
+            LOGGER.warn(MessageCodes.AUTH_024, manifestPath, details.getMessage());
```

Comment:
Is the use of version in Hauth important enough to warrant an error?

Reply:
Also, regarding "Remove HAUTH_VERSION from pom" -- it doesn't look like the POM was updated in this PR(?)

Reply:
Per the README, HAUTH_VERSION is a value, and `Config.HAUTH_VERSION` is referenced in `AccessTokenHandler`, `AccessCookieHandler`, `AccessCookieServiceImpl`, and `SinaiAccessTokenHandler`.

Reply:
So, should it be more than a warning? Seems, to me, that missing something that's required should be an error, perhaps also bubbling up a failed future?

Reply:
I was thinking that it would be nice (as an enhancement) for HTTP responses to include the [Server](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server) header, which I'd see looking something like `Hauth/x.y.z`. [Here's how prl-harvester deals with this](https://github.com/UCLALibrary/prl-harvester/blob/614d5e855ec27ce4a9d1441aa389132d9341e4ac/src/main/java/edu/ucla/library/prl/harvester/Config.java#L144-L160) (MDN says User-Agent and Server are usually in a similar format). However, simply the product name is also acceptable, so it's up to us on how strict we want to be with the value.

Reply:
I like your suggestion @ksclarke

Reply:
Can the function passed to `ConfigRetriever.setConfigurationProcessor` (here `setAppVersion`) return a Future? And if so, does that require changing how `setConfigurationProcessor` is handled?

Reply:
Yes, `setAppVersion` could return a future, but you're right you'd have to handle that instead of just passing the result to `setConfigurationProcessor` since that takes a function that returns a `JsonObject`: `setConfigurationProcessor(Function<JsonObject, JsonObject> processor);`.